### PR TITLE
Add Win11 helix-matrix/quarantine queue, remove win7/8

### PR DESCRIPTION
**Pull request by wtgodbe**
Part of https://github.com/dotnet/aspnetcore/issues/36032, https://github.com/dotnet/aspnetcore/issues/34741

### CODE REVIEW

**Comment by wtgodbe** [comment]
Queued https://dev.azure.com/dnceng/public/_build/results?buildId=1334918&view=results

**Review by HaoK — APPROVED** [review]
Looks good assuming the run doesn't crash and burn :)

**Comment by HaoK** [comment]
Usually the only thing to watch out for when adding new queues is we have some tests that are skipped on specific queues by name, which will fail whenever we add new ones, but generally those are linux / macOS

**Comment by Tratcher** [comment]
Have you run the tests locally recently? I'd expect some failures caused by recent runtime updates that we haven't addressed.

**Review by dougbu — APPROVED** [review]
Don't we have some Linux changes e.g. moving to a later version of Debian❔ Suggest we clean up the whole mess at once but it's your call.

**Comment by wtgodbe** [comment]
> Have you run the tests locally recently?

Not all of them, but for now they're only running in the quarantine job so that shouldn't be blocking (unless there are unquarantined Quic tests?)

**Comment by wtgodbe** [comment]
@premun @MattGal getting an error with the new queue:

>  Sending Job to Helix-Client-Enterprise-Win11-Preview-08-30-2021...
D:\workspace\_work\1\s\.packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21427.6\tools\Microsoft.DotNet.Helix.Sdk.MonoQueue.targets(55,5): error : ArgumentException: Unknown QueueId. Check that authentication is used and user is in correct groups. [D:\workspace\_work\1\s\eng\helix\helix.proj]

**Comment by MattGal** [comment]
> @premun @MattGal getting an error with the new queue:
> 
> > Sending Job to Helix-Client-Enterprise-Win11-Preview-08-30-2021...
> > D:\workspace_work\1\s.packages\microsoft.dotnet.helix.sdk\6.0.0-beta.21427.6\tools\Microsoft.DotNet.Helix.Sdk.MonoQueue.targets(55,5): error : ArgumentException: Unknown QueueId. Check that authentication is used and user is in correct groups. [D:\workspace_work\1\s\eng\helix\helix.proj]

Taking a peek.

**Comment by MattGal** [comment]
@wtgodbe can you please try `Windows.11.Amd64.ClientPre.Open`? ([it exists](https://helix.dot.net/api/info/queues/Windows.11.Amd64.ClientPre.Open?api-version=2019-06-17))

File: eng/targets/Helix.Common.props [hunk 1]
```diff
@@ -43,9 +43,7 @@
   <!-- x64 Queues for public helix-matrix.yml and quarantine pipelines, Windows cases-->
   <ItemGroup Condition="'$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' == 'true' AND '$(_UseHelixOpenQueues)' == 'true'">
     <HelixAvailableTargetQueue Include="Windows.10.Amd64.Server20H2.Open" Platform="Windows" />
-    <!-- TODO Re-enable Win-7 queue when dotnet restore are fixed. https://github.com/dotnet/aspnetcore/issues/32683 -->
-    <!-- <HelixAvailableTargetQueue Include="Windows.7.Amd64.Open" Platform="Windows" /> -->
-    <HelixAvailableTargetQueue Include="Windows.81.Amd64.Open" Platform="Windows" />
+    <HelixAvailableTargetQueue Include="Helix-Client-Enterprise-Win11-Preview-08-30-2021" Platform="Windows" />
```
[the remark this answers is not in the document]
Comment (MattGal) [reply eng/targets/Helix.Common.props:46]: correct, was trying to make one of those commit suggestions to say this but failed :)

**Comment by wtgodbe** [comment]
https://dev.azure.com/dnceng/public/_build/results?buildId=1335689&view=results

**Comment by wtgodbe** [comment]
IIS Cert failures on Win11:

> "PS: Running 'C:\h\w\BCEE0A09\w\A66008D4\e\UpdateIISExpressCertificate.ps1' "

Major  Minor  Build  Revision
-----  -----  -----  --------
10     0      22000  65      
Certificate "github.com/aspnet/AspNetCore IIS testing cert" added to store.

CertUtil: -importPFX command completed successfully.

SSL Certificate deletion failed, Error: 2
The system cannot find the file specified.

**Comment by wtgodbe** [comment]
This is the same as we were seeing when we introduced https://github.com/dotnet/aspnetcore/pull/34726 (CC @Tratcher @MattGal), which is surprising as the queue is Win11. Unless the IIS Test Cert script is called before sending the job to helix, which certainly seems possible as it's part of `HelixPreCommand`

**Comment by wtgodbe** [comment]
Also the block I modified apparently doesn't effect quarantine-pr jobs, only the nightly quarantine runs, which is probably fine for getting coverage for now. Test job: https://dev.azure.com/dnceng/public/_build/results?buildId=1338195&view=results

**Comment by wtgodbe** [comment]
The IIS thing is the same that we saw when the Win10 pool we used got upgraded: https://github.com/dotnet/core-eng/issues/13582#issuecomment-886854042. The solution there was "please revert the upgrade", but I guess here we need to skip all of these tests on Win11 since it is necessarily "too new" - does that sound right @Tratcher?

**Comment by Tratcher** [comment]
The issue is in the project setup scripts, you can't even skip the tests, you'd have to skip whole projects.

Can you try to repro this locally so we can figure out why the script is failing? Is IIS Express not installed on this agent?

**Comment by Tratcher** [comment]
Looking at the script again, it seems to be this line failing?
https://github.com/dotnet/aspnetcore/blob/cc5f8968e74c9ea2224b578d269e4bb3a2b0b085/src/Servers/IIS/tools/UpdateIISExpressCertificate.ps1#L22
That would be expected to fail if IIS Express wasn't installed.

Also note this isn't exactly the same error as before, before it failed on add, not delete.

**Comment by wtgodbe** [comment]
Yeah, it works for me locally on Win11. @premun @MattGal is IIS Express not installed on the Win11 machines?

**Comment by MattGal** [comment]
> Yeah, it works for me locally on Win11 (when ran as admin). @premun @MattGal is IIS Express not installed on the Win11 machines?

Checking out [the definition](https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines?path=%2Fdefinitions%2Fshared%2Fwindows.yaml&version=GBproduction&line=1064&lineEnd=1075&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) I see only [Windows10Base](https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines?path=%2Fdefinitions%2Fshared%2Fwindows.yaml&version=GBproduction&line=116&lineEnd=130&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) so unless it were preinstalled on the OS there's definitely no IISExpress here.

Checking the original issue (https://github.com/dotnet/core-eng/issues/13582) I don't see that being asked for... feel free to create a dotnet/core-eng issue with your list of requirements and we'll get onto it.

**Comment by wtgodbe** [comment]
It should be the same requirements as the Win10 machines where we run our IIS tests today, @Tratcher do you remember if we had to get specially configured machines for those?

**Comment by Tratcher** [comment]
Right, same requirements as the other windows agents.

The only extra thing we'll consider adding later is the Http3Enabled reg key.

**Comment by wtgodbe** [comment]
Looks like the 20h1 and 20h2 images enable IIS explicitly, but the Windows10-base image doesn't, so we'd want it explicitly enabled for the 11 image as well. I filed https://github.com/dotnet/core-eng/issues/14275 to track that work

**Comment by Tratcher** [comment]
IIS and IIS Express are separate. IIS Express is installed by VS, or as a standalone app.

**Comment by wtgodbe** [comment]
I misspoke, the 20h1 and 20h2 images install IIS Express: https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines?path=%2Fdefinitions%2Fshared%2Fwindows.yaml&version=GBproduction&line=982&lineEnd=983&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

**Comment by MattGal** [comment]
> I misspoke, the 20h1 and 20h2 images install IIS Express: https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines?path=%2Fdefinitions%2Fshared%2Fwindows.yaml&version=GBproduction&line=982&lineEnd=983&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Understood, please review the artifacts there if you can and let me know if there are any other differences you depend on.

**Comment by wtgodbe** [comment]
I think that's the only addition we need, I don't believe we need `windows-enable-tls13` since it's already on by default in Win11

**Comment by wtgodbe** [comment]
Update is now rolled out

**Comment by wtgodbe** [comment]
And here's helix-matrix: https://dev.azure.com/dnceng/public/_build/results?buildId=1349415&view=results

**Comment by wtgodbe** [comment]
For quarantine validation: https://dev.azure.com/dnceng/public/_build/results?buildId=1350161&view=results

**Comment by wtgodbe** [comment]
Tests are running now, but we've got some Http3 failures on Win11 (CC @JamesNK @Tratcher)

> Interop.FunctionalTests.Http3.Http3RequestTests.POST_ServerAbort_ClientReceivesAbort(protocol: Http3)

> Assert.Equal() Failure
Expected: RequestCancelled
Actual:   InternalError
at Interop.FunctionalTests.Http3.Http3RequestTests.POST_ServerAbort_ClientReceivesAbort(HttpProtocols protocol) in /_/src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:line 441

> Interop.FunctionalTests.Http3.Http3RequestTests.GET_GracefulServerShutdown_AbortRequestsAfterHostTimeout(protocol: Http2)

> Assert.Equal() Failure
↓ (pos 4)
Expected: The connection was aborted because the server···
Actual:   The HTTP/2 connection faulted.
↑ (pos 4)
at Interop.FunctionalTests.Http3.Http3RequestTests.GET_GracefulServerShutdown_AbortRequestsAfterHostTimeout(HttpProtocols protocol) in /_/src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:line 1409

> Interop.FunctionalTests.Http3.Http3RequestTests.GET_ServerAbort_ClientReceivesAbort(protocol: Http3)

> Assert.Equal() Failure
Expected: RequestCancelled
Actual:   InternalError
at Interop.FunctionalTests.Http3.Http3RequestTests.GET_ServerAbort_ClientReceivesAbort(HttpProtocols protocol) in /_/src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs:line 498

> Interop.FunctionalTests.Http3.Http3TlsTests.ClientCertificate_AllowOrRequire_Available_Invalid_Refused(mode: AllowCertificate & mode: RequireCertificate)

> Assert.IsType() Failure
Expected: System.Net.Quic.QuicOperationAbortedException
Actual:   (null)
at Interop.FunctionalTests.Http3.Http3TlsTests.ClientCertificate_AllowOrRequire_Available_Invalid_Refused(ClientCertificateMode mode) in /_/src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3TlsTests.cs:line 189

**Comment by wtgodbe** [comment]
@JamesNK @Tratcher @adityamandaleeka any concerns with this PR? If not I can merge, backport, and open issues for the failing tests.

**Comment by Tratcher** [comment]
I'm concerned that none of those failures repro locally, but they're all quarantined right?

**Review by Tratcher — APPROVED** [review]

**Pull request merged**